lib/widgets/message_list.dart

```diff
@@ -65,11 +65,26 @@ class _MessageListState extends State<MessageList> {
         style: const TextStyle(color: Color.fromRGBO(0, 0, 0, 1)),
         child: ColoredBox(
             color: Colors.white,
-            child: Padding(
-                padding: const EdgeInsets.symmetric(horizontal: 8),
-                child: ConstrainedBox(
-                    constraints: const BoxConstraints(maxWidth: 760),
-                    child: _buildListView(context)))));
+            child: Center(
+              // Pad the left and right insets, for small devices in landscape.
+              child: SafeArea(
+                  // A non-ancestor (the compose box) pads the bottom inset;
+                  // prevent extra padding here.
+                  bottom: false,
+
+                  // A non-ancestor (the app bar) pads the top inset. But no
+                  // need to prevent extra padding with `top: false`, because
+                  // Scaffold, which knows about the app bar, sets `body`'s
+                  // ambient `MediaQueryData.padding` to have `top: 0`:
+                  //   https://github.com/flutter/flutter/blob/3.7.0-1.2.pre/packages/flutter/lib/src/material/scaffold.dart#L2778
+
+                  // Keep some padding when there are no horizontal insets,
+                  // which is usual in portrait mode.
+                  minimum: const EdgeInsets.symmetric(horizontal: 8),
```

Comment:
What's the reasoning for doing this rather than using a `Padding` in the child?

IIUC, the difference is that when there is an inset (of 8px or more), using `minimum` means we won't add padding beyond the inset. But in your screenshot it looks like there is padding there in the end:
![image](https://user-images.githubusercontent.com/28173/214124707-ef509360-0141-4229-8490-f40c485dcdc6.png)
Is that coming from somewhere deeper in the tree?

Reply:
Do you mean the ~8px between the outline of a message in the message list, and that black oval hardware thing? That's part of the inset, as told to us by the platform, and the screenshot doesn't show any padding on top of that.

<img width="91" alt="image" src="https://user-images.githubusercontent.com/22248748/214130385-89781670-4e11-4ff5-a8b0-114dfe7cf126.png">

One thing that might help this make sense: the "x" coordinate of the right edge of the message list is where the screen starts rounding its corners, so from that "x" coordinate onward, you want to be protected from rendering content that might get clipped by that rounding.

Or another: it looks like that same ~8px distance is used to offset the oval from the right edge of the screen, which has the nice effect of centering the oval in this area that apps are asked to treat specially.

Reply:
Here are screenshots of the added `SafeArea` being highlighted, and then its child:

<img width="1321" alt="image" src="https://user-images.githubusercontent.com/22248748/214134367-2580c0e1-484d-464a-8047-1322f403f5f8.png">

<img width="1428" alt="image" src="https://user-images.githubusercontent.com/22248748/214134080-42b8ec5b-9644-4699-8ff4-e1302100dd0c.png">